nntrainer/layers/embedding.cpp

```diff
@@ -52,49 +60,14 @@ void EmbeddingLayer::finalize(InitLayerContext &context) {
 }
 
 void EmbeddingLayer::setProperty(const std::vector<std::string> &values) {
-  /// @todo: deprecate this in favor of loadProperties
-  for (unsigned int i = 0; i < values.size(); ++i) {
-    std::string key;
-    std::string value;
-    std::stringstream ss;
-
-    if (getKeyValue(values[i], key, value) != ML_ERROR_NONE) {
-      throw std::invalid_argument("Error parsing the property: " + values[i]);
-    }
-
-    if (value.empty()) {
-      ss << "value is empty: key: " << key << ", value: " << value;
-      throw std::invalid_argument(ss.str());
-    }
-
-    /// @note this calls derived setProperty if available
-    setProperty(key, value);
-  }
-}
-
-void EmbeddingLayer::setProperty(const std::string &type_str,
-                                 const std::string &value) {
-  using PropertyType = nntrainer::Layer::PropertyType;
-  int status = ML_ERROR_NONE;
-  nntrainer::Layer::PropertyType type =
-    static_cast<nntrainer::Layer::PropertyType>(parseLayerProperty(type_str));
-
-  switch (type) {
-  case PropertyType::in_dim: {
-    status = setUint(in_dim, value);
-    throw_status(status);
-  } break;
-  case PropertyType::out_dim: {
-    status = setUint(out_dim, value);
-    throw_status(status);
-  } break;
-  default:
-    LayerImpl::setProperty(type_str, value);
-    break;
-  }
+  auto remain_props = loadProperties(values, embedding_props);
+  LayerImpl::setProperty(remain_props);
 }
 
 void EmbeddingLayer::forwarding(RunLayerContext &context, bool training) {
+  unsigned int in_dim = std::get<props::InDim>(embedding_props);
+  unsigned int out_dim = std::get<props::OutDim>(embedding_props);
```

Comment:
i recommend to use get input and output dimension from input_ and hidden_ itself.
As this can be out of scope for this PR, adding a TODO works too.

Reply:
can you also add a TODO to support setBatch for EmbeddingLayer.

Reply:
Done